src/index.d.ts

```diff
@@ -0,0 +1,63 @@
+import appdmg = require('appdmg');
+
+declare module 'electron-installer-dmg' {
+  namespace createDMG {
+      interface Content {
+          path: string;
+          type: 'link' | 'file';
+          x: number;
+          y: number;
+      }
+
+      interface CreateOptions {
+          /**
+           * Additional options to pass through to [`appdmg`](https://npm.im/appdmg).
+           * You can use this to set additional features like `background-color` and `code-sign`.
+           * See the docs of the `appdmg` module for all possible options.
+           */
+          additionalDMGOptions?: Partial<appdmg.Specification>;
+          /**
+           * The `.app` directory generated by
+           * [electron-packager](https://github.com/electron-userland/electron-packager).
+           */
+          appPath: string;
+          /** Path to the background for the DMG window. Background image should be of size 658×498. */
+          background?: string;
+          /**
+           * The content that will appear in the window when user opens the .dmg file.
+           * [Default: Array of two icons, application and application destination folder].
+           */
+          contents?: Content[] | ((opts?: CreateOptions) => Content[]);
+          /** Enable debug message output. */
+          debug?: boolean;
+          /**
+           * @deprecated
+           * The directory to put the DMG into. [Default: `process.cwd()`].
+           */
+          dmgPath?: string;
+          /** Disk image format. [Default: `UDZO`]. */
+          format?: 'UDRW' | 'UDRO' | 'UDCO' | 'UDZO' | 'UDBZ' | 'ULFO';
+          /** Path to the icon to use for the app in the DMG window. */
+          icon?: string;
+          /** How big to make the icon for the app in the DMG. [Default: `80`]. */
+          iconSize?: number;
+          /** The application name. */
+          name: string;
+          /**
+           * @deprecated
+           * The application name.
+           */
+          productName?: string;
+          /** The directory to put the DMG into. [Default: `process.cwd()`]. */
+          out?: string;
+          /** Overwrite an existing DMG file if if already exists. */
+          overwrite?: boolean;
+          /** The title of the produced DMG, which will be shown when mounted. */
+          title?: string;
+      }
+  }
+
+  function createDMG(createOptions: createDMG.CreateOptions): Promise<createDMG.CreateOptions>;
```

Comment:
This needs documentation.

Reply:
Where should I take this from? For this function, there is no documentation at all in this repository.